File: solutions-reviews/python/dojo/pm/fizbuzz/FizzBuzz.py
#Write a program that prints the numbers from 1 to 100. But for multiples of three print "Fizz" instead of the number and for the multiples of five print "Buzz". For numbers which are multiples of both three and five print "FizzBuzz".

def FizzBuzz(length):
    result = []
    if length < 0:
        raise ValueError
    for i in range(1, length + 1):
        if i % 15 == 0:
            result.append('FizzBuzz')
        elif i % 3 == 0:
            result.append('Fizz')
        elif i % 5 == 0:
            result.append('Buzz')
        else:
            result.append(i)
    return result

#FizzBuzz(30)

File: solutions-reviews/javascript/dojo/pm/fizzbuzz/fizzbuzz.js
'use strict';

function fizzbuzz(index) {
  var result = [];
  var i = 1;
  while (i < 101) {
    if (i % 15 === 0) {
      result.push('fizzbuzz');
    } else if (i % 3 === 0) {
      result.push('Fizz');
    } else if (i % 5 === 0) {
      result.push('Buzz');
    } else {
      result.push(i);
    }
    i++; 
  }
  if (typeof index === 'undefined'){
    return result;
  }
  return result[index-1];
}; 

module.exports = fizzbuzz;
